### DISCUSSION TITLE
Lesson 9: Contract was verified on etherscan but "Unable to verify if this is an Automation compatible contract." when put in to "Register new Upkeep"

### DISCUSSION THREAD

**Discussion by DuyVo96** [Cyfrin/foundry-full-course-cu#153] [General]
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/85976363/60777971-7fcd-4a00-9bfa-b9f2b614da04)

This is my contract: https://sepolia.etherscan.io/address/0xcf16afaa9c530b8a441a240ff5ed8ea8f3eae38f#readContract.

Can someone help me figure out what the issue is?

**Comment by ghost** [comment]
Hello @DuyVo96, you haven't inherited `AutomationCompatible.sol`. Check this [example](https://docs.chain.link/chainlink-automation/compatible-contracts#example-contract) from the Chainlink docs.

**Reply by DuyVo96** [reply]
Ah I updated AutomationCompatible.sol and it's ok but now I got new issue with Perfom Upkeep, after Create & Fund Upkeep it did not auto run.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/85976363/7985d457-4abd-4423-a48f-b88c260e2406)

**Reply by DuyVo96** [reply]
I fixed it!  Thanks for supporting me.

**Comment by cromewar** [comment]
@DuyVo96 Please share how you solved this so it might be helpful for the community.

**Reply by gofotog** [reply]
@DuyVo96 please share your solved. same issue

**Reply by Eshan-Sharma** [reply]
I was facing the same issue, ```copy addresss``` worked for me

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/43044334/4e273ff2-4c9b-40b1-ac60-4b85ed2cf812)
